File: src/algorithms/k-length-apart.js
/**
 * Takes an integer array nums and an integer k, returning true if all 1's are
 * at least k places away from each other. Otherwise, returns false.
 * 
 * @param {number[]} nums - An array of the numbers 0 and 1
 * @param {number} k - A number of spaces 1's should be away from each other
 * @returns {boolean} - Returns true if all 1's are at least k spaces apart
 */
function kLengthApart (nums, k) {
  for (let i = 0; i < nums.length; i++) {
    let placesAway = 0;
    if (nums[i] === 1) {
      while (placesAway < k) {
        i++;
        if (nums[i] !== 1) {
          placesAway++;
        } else {
          return false;
        }
      }
    }
  }
  return true;
}

File: src/algorithms/k-length-apart.py
def kLengthApart(nums, k):
    """Given an binary array nums and an integer k, return true if all 1's are 
    at least k places away from each other, otherwise return false.

    Args:
        nums (list[int]): A list of integers.
        k (int): A positive integer.

    Returns:
        Boolean: Returns True if all 1's are at least k places away.
    """
    lastSeen = None
    index = 0
    while index < len(nums):
        if nums[index] == 1:
            if lastSeen != None and index - lastSeen <= k:
                return False
            lastSeen = index
        index += 1
    return True
